### ISSUE TITLE
Add auto installer for npm and for pm2 for the interactive cli installer (for local installs)

### ISSUE THREAD

**Issue by MozerBYU** [louislam/uptime-kuma#3646]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

Other

### 🔖 Feature description

Basically while doing the auto installer, it fails with an error that pm2 couldn't be installed. Doing a bit of digging I found that npm wasn't installed and that the auto installer doesn't attempt to install npm to rectify the issue.

I feel it could be beneficial to have it auto install npm and the pm2 package. So that the installer is more hands-off and less user-error prone.

I'd be more than happy to submit a pull request if you'd like that accomplishes the above.

### ✔️ Solution

Basically, have the installer check for an npm installation before trying to install the pm2 package. If one doesn't exist, go ahead and install npm for the user. Then continue to proceed with the pm2 package installation.

**Comment by MozerBYU** [comment]
Clarified some language in the feature request.

**Comment by CommanderStorm** [comment]
We have added help texts to the installer in https://github.com/louislam/uptime-kuma/pull/3603
Note that the installer has to run on both linux/mac + windows => installing npm via the guide we now link is eazier and less likely to result in serious/hard to debug defects imo

**Comment by CommanderStorm** [comment]
Do I read you correctly that you have node, but not npm installed?
What was the error message shown?

**Comment by jult** [comment]
Maybe get it to run as snaps or similar? UK is one of the least reliable packages on my home debian cluster, not exactly what you'd expect from a monitoring server-package. The dependencies with pm2, npm, node-js, and a reverse proxy are a multitude of points of failure, you'll need an entire system to watchdog and monitor your uptime-kuma.

**Comment by louislam** [comment]
> running the auto installer on the Readme

I think I will remove the installer later, because it is really hard to maintain such things in Linux. Normally `npm` will be installed along with Node.js, but in some cases, it doesn't.

> Maybe get it to run as snaps or similar

May need someone to do that, I am not interested in it. As far as I know, someone made it for Arch Linux (https://aur.archlinux.org/packages/uptime-kuma)


> The dependencies with pm2, npm, node-js, and a reverse proxy are a multitude of points of failure, you'll need an entire system to watchdog and monitor your uptime-kuma.

Sorry I don't get what you mean here. Are you expecting a JavaScript application running without Node.js and npm? And are you expecting to "reverse-proxy" without a reverse proxy?

Talk is cheap. If you think you can write this app in a single executable file without dependencies. Just show it to the world.

**Comment by jult** [comment]
I've switched to [ntopng](https://packages.ntop.org/) (the free community edition). May look overwhelming at first, but it's way more sophisticated (first created in 1998), and once you get the hang of adding alerts and pings, it's super logical, and your dashboard is entirely configurable.
